File: cs/type-safe-builder-test/BuilderTest.cs
using NUnit.Framework;
using type_safe_builder;

namespace Tests
{
	public class Tests
	{
		[Test]
		public void TestItShouldBuildAPoint()
		{
			var point = Point3DBuilderExtensions.Point3D
				.Z(3.0)
				.Y(2.0)
				.X(1.0)
				.Build();

			var expected = new Point3D(1.0, 2.0, 3.0);

			Assert.AreEqual(expected, point);
		}
	}
}

File: java/lib/src/test/java/type_safe_builder/BuilderTest.java
package type_safe_builder;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import static type_safe_builder.Point3DBuilderFunctions.*;

class BuilderTest {
	@Test
	public void testItShouldBuildAPoint() {
		var point = build(x(y(z(point3d(), 3.0), 2.0), 1.0));
		var expected = new Point3D(1.0, 2.0, 3.0);

		assertEquals(expected, point);
	}
}
